src/main/java/teammates/ui/controller/Action.java

```diff
@@ -701,4 +701,14 @@ private boolean isMasqueradeModeRequested(AccountAttributes loggedInUser, String
     protected void excludeStudentDetailsFromResponseParams() {
         regkey = null;
     }
+
+    protected void appendToStatus(String message) {
```

Comment:
I don't think this method is necessary. Instead, modify all the places that use it such that they only call `statusToAdmin.add` once, after the entire message has been prepared.